pkg/reconciler/taskrun/resources/pod.go

```diff
@@ -275,6 +298,43 @@ func MakePod(images pipeline.Images, taskRun *v1alpha1.TaskRun, taskSpec v1alpha
 			}
 		}
 
+		// If the step specifies a Script, generate and invoke an
+		// executable script file containing each item in the script.
+		if s.Script != "" {
+			placeScripts = true
+			// Append to the place-scripts script to place the
+			// script file in a known location in the scripts volume.
+			tmpFile := filepath.Join(scriptsDir, names.SimpleNameGenerator.RestrictLengthWithRandomSuffix(fmt.Sprintf("script-%d", i)))
+			// NOTE: quotes around 'EOF' are important. Without
+			// them, ${}s in the file are interpreted as env vars
+			// and likely end up replaced with empty strings. See
+			// https://stackoverflow.com/a/27921346
+			placeScriptsStep.Args[1] += fmt.Sprintf(`tmpfile="%s"
+touch ${tmpfile} && chmod +x ${tmpfile}
+cat > ${tmpfile} << 'EOF'
+%s
```

Comment:
Hi! One question for you. It seems like this would forbid users from using heredocs delimited by EOF in their own scripts, likely resulting a weird error in this generator container if they attempt to. Would it make sense to either (a) warn/error if `s.Script` contains a line that starts with `EOF`, or (b) write the scripts to a config-map and mount them into the step containers with the executable bit set?

Reply:
Excellent point!

I'd like to fix this in a followup (tomorrow hopefully) whereby the heredoc isn't `EOF`, it's a randomly generated string, possibly a hash of the step name or something so that we can validate it upfront and reject people who happen to trip over it.

I'd feel a little bad creating and deleting ConfigMaps all the time (or failing to delete them and having them clog things up). It might be a worthwhile alternative if the heredoc hack doesn't pan out.

Thanks for catching this, let me know what you think, or if there are any other options we might consider.

Reply:
That seems totally reasonable to me. I did think of two other alternatives, neither of which I think are preferable to randomly (or deterministically) generating the heredoc:
1) The old-hat trick of embedding the data at the end of the shell script (after an `exit`), often used by shell installer scripts.
2) Shell-escaping the script and using `<<<'the-escaped-script'` to write it to the file. This could potentially look crazy and I'd worry about the quality of the sanitization, and I'm not aware of any standard library function to do the escaping.

Reply:
Here's the followup I intend to send after this merges: https://github.com/ImJasonH/pipeline/compare/script2...ImJasonH:script-here?expand=1